packages/core/content-manager/server/services/entity-manager.js

```diff
@@ -266,6 +266,84 @@ module.exports = ({ strapi }) => ({
     return mappedEntity;
   },
 
+  async publishMany(entities, uid) {
+    if (!entities.length) {
+      return null;
+    }
+
+    // Validate entities before publishing, throw if invalid
+    await Promise.all(
+      entities.map((entityToUpdate) => {
+        if (entityToUpdate[PUBLISHED_AT_ATTRIBUTE]) {
+          throw new ApplicationError('already.published');
```

Comment:
I'm not sure I understand the need to throw an error for this. Couldn't we just skip the already published entites?

Reply:
I was thinking we should mimic the behavior of the publish/unpublish methods.

I thought this error actually logged something in the server logs, but it looks like it doesn't. I think that could be useful in case someone tried to programatically publish a mix of published and draft entities, 

If we skip it instead of throwing, do you think on the frontend we should show the publish button even if a published entity is selected?

Reply:
I think it's a bit different, because for a single entity publish, nothing was gonna happen anyway if you tried to re-publish it. But here it means you cancel all other entity publications

I can't find the place where I saw that but I thought I remembered we would only hide the publish/unpublish buttons if 100% of the selected entries were published/unpublished, not when it was a mix of the 2. It's probably safer to double check with Yannis

Reply:
>  I think it's a bit different, because for a single entity publish, nothing was gonna happen anyway if you tried to re-publish it.

Not sure I completely understand, but for publish if this error wasn't thrown and the user was able to trigger another publish action on an already published entity then the PUBLISHED_AT date would be updated.

> But here it means you cancel all other entity publications
Isn't that a good thing since the user didn't mean to re-publish something? How will they know that is was skipped.